client/src/pages/Feed.js

```diff
@@ -1,17 +1,74 @@
-import React from "react";
+import React, { useState } from "react";
 import styled from "styled-components";
-
-import FilterBox from "../components/FilterBox/FilterBox";
+import filterOptions from "../assets/data/filterOptions";
+import fakePosts from "../assets/data/fakePosts";
+import FilterBox from "../components/Feed/FilterBox";
+import Posts from "../components/Feed/Posts";
 
 const FeedWraper = styled.div`
   width: 100%;
   padding: 20px 0;
 `;
 
 const Feed = () => {
+  const [modal, setModal] = useState(false);
+  const [activePanel, setActivePanel] = useState("");
+  const [selectedFilters, setSelectedFilters] = useState({});
+  const [location, setLocation] = useState("");
+
+  const handleModal = (isOpen, panelIdx) => (e) => {
+    e.preventDefault();
+    setModal(isOpen);
+    setActivePanel(panelIdx > -1 ? `${panelIdx}` : null);
+  };
+
+  const handleQuit = (e) => {
+    e.preventDefault();
+    setModal(false);
+    setActivePanel(null);
+    setSelectedFilters({});
+    setLocation("");
```

Comment:
it could be nice to work with "initial state" and reset to that, it used to be a React pattern few years ago with `initialState()` and then there were more class components but now with hooks I am not sure if the best way is to use one initial state object or have a separate hook for every state property, seems it could also work better with `userReducer` https://reactjs.org/docs/hooks-reference.html#usereducer

Reply:
Good point! I haven't used `userReducer` before but it sounds like a good fit. There's also a pattern to use dynamic hooks for specific functionalities like toggling state or handling input. I'll look into the options